tests/nnstreamer_filter_llamacpp/unittest_filter_llamacpp.cc

```diff
@@ -0,0 +1,371 @@
+    g_object_set (G_OBJECT (appsink), "emit-signals", TRUE, NULL);
+    g_signal_connect (appsink, "new-sample", G_CALLBACK (new_sample_cb), NULL);
+  }
+
+  /**
+   * @brief Push data into the pipeline.
+   */
+  void data_push (const gchar *data)
+  {
+    gsize data_size;
+    GstBuffer *buffer;
+    GstMapInfo map;
+
+    data_size = strlen (data);
+    buffer = gst_buffer_new_allocate (NULL, data_size, NULL);
+
+    gst_buffer_map (buffer, &map, GST_MAP_WRITE);
+    memcpy (map.data, data, data_size);
+    gst_buffer_unmap (buffer, &map);
+
+    EXPECT_EQ (gst_app_src_push_buffer (GST_APP_SRC (appsrc), buffer), GST_FLOW_OK);
+  }
+};
+
+/**
+ * Note. Inference libraries like LLP and llama.cpp generally do not allow
+ * modification of such settings at runtime after the context (e.g., model,
+ * session, etc.) has been created. In particular, parameters like n_gpu_layers
+ * must be configured at model loading time and cannot be changed afterwards.
+ */
+
+/**
+ * @brief Test case for tensor_filter llama-cpp plugin with singleInputMultipleOutputsAsync_p
+ */
+TEST_F (NNStreamerFilterLlamaCppTest, singleInputMultipleOutputsAsync_p)
+{
+  gboolean invoke_async = TRUE;
+  gint num_predict = 10;
+  if (skip_test) {
+    GTEST_SKIP ()
+        << "Skipping singleInputMultipleOutputsAsync_p test due to missing model file"
+           "Please download model file from https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF";
+  }
+  ASSERT_TRUE (this->loaded);
+
+  new_sample_count = 0;
+  create_pipeline (model, invoke_async, num_predict);
+  data_push ("Hello my name is");
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (3000000);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  EXPECT_GT (new_sample_count, 10);
+}
+
+/**
+ * @brief Test case for tensor_filter llama-cpp plugin with singleInputSingleOutputSync_p
+ */
+TEST_F (NNStreamerFilterLlamaCppTest, singleInputSingleOutputSync_p)
+{
+  gboolean invoke_async = FALSE;
+  gint num_predict = 10;
+  if (skip_test) {
+    GTEST_SKIP ()
+        << "Skipping singleInputSingleOutputSync_p test due to missing model file"
+           "Please download model file from https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF";
+  }
+  ASSERT_TRUE (this->loaded);
+
+  new_sample_count = 0;
+  create_pipeline (model, invoke_async, num_predict);
+  data_push ("Hello my name is");
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (3000000);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  EXPECT_EQ (new_sample_count, 1);
+}
+
+/**
+ * @brief Test case for tensor_filter llama-cpp plugin with multipleInputsMultipleOutputsAsync_p
+ */
+TEST_F (NNStreamerFilterLlamaCppTest, multipleInputsMultipleOutputsAsync_p)
+{
+  gboolean invoke_async = TRUE;
+  gint num_predict = 10;
+  if (skip_test) {
```

Comment:
Please refer to [this review comment](https://github.com/nnstreamer/nnstreamer/pull/4739/files#r2231343182).

Reply:
Thank you.